src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -94,6 +94,20 @@ public TimeSpan UserProcessorTime
             }
         }
 
+        /// <summary>Gets parent process ID</summary>
+        private int? ParentProcessId
+        {
+            get
+            {
+                EnsureState(State.HaveNonExitedId);
+
+                // Returns null if permissions do not allow querying the specified process
```

Comment:
Seems like the Windows version of ParentProcessId throws Win32Exception when permissions are insufficient, whereas the OSX one returns null. I guess that's OK as they're private.

Reply:
Thanks, @danmosemsft! For consistency, changed to throw an exception instead of return null.